main/src/library/Fixpoint3/Phase/RenamePredSyms.flix

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+/// For the rest of the compiler we expect a PredSym, `PredSym(name, id)`, to have unique `id`.
+/// This phase passes over the Datalog-Ast and replaces the id and after everything is done
+/// recreates the old `PredSym`s for the resulting model.
+/// 
+/// It also contains method for recreating the original PredSyms such that the model given to Flix
+/// is correct.
+//
+mod Fixpoint3.Phase.RenamePredSyms {
+    use Fixpoint.Ast.Datalog.{Datalog, Constraint, BodyPredicate, BodyTerm, HeadTerm, Polarity, VarSym}
+    use Fixpoint.Ast.Datalog.Datalog.{Datalog, Model, Join}
+    use Fixpoint.Ast.Datalog.Constraint.Constraint
+    use Fixpoint.Ast.Datalog.HeadPredicate
+    use Fixpoint.Ast.Datalog.HeadPredicate.HeadAtom
+    use Fixpoint.Ast.Datalog.BodyPredicate.{BodyAtom, Functional, Guard1, Guard2, Guard3, Guard4, Guard5}
+    use Fixpoint3.Ast.Ram.RelSym
+    use Fixpoint.Ast.Shared.{Denotation, isRelational, PredSym}
+    use Fixpoint.Ast.Shared.PredSym.PredSym
+    use Fixpoint3.UniqueInts
+    use Fixpoint3.Util.getOrCrash
+
+    type alias FixInformation = Map[Int32, PredSym]
+
+    ///
+    /// Maps all `PredSym`s in `d` to have unique ids. Returns the resulting
+    /// program and information which `renamePredSymToFlixStyle` can use to recreate the original program.
+    ///
+    @Internal
+    pub def renamePredSyms(rc: Region[r], d: Datalog): (Datalog, FixInformation) \ r =
+        let mapping = UniqueInts.empty(rc);
+        let fixed = renamePredSymsInternal(d, mapping);
+        (fixed, UniqueInts.reverse(mapping))
+
+    ///
+    /// Recursively applies `mapping` to the different Datalog enums to make `PredSym`s unique.
+    ///
+    def renamePredSymsInternal(d: Datalog, mapping: UniqueInts[PredSym, r]): Datalog \ r = match d {
+        case Datalog(facts, rules) =>
+            let transformer = mapForward(mapping);
+            Datalog(remapConstraints(facts, transformer), remapConstraints(rules, transformer))
+        case Model(facts) =>
+            Model(Map.foldLeftWithKey(acc -> k -> v -> match k {
+                case RelSym.Symbol(predSym, arity, den) =>
+                    let newSym = RelSym.Symbol(mapForward(mapping, predSym), arity, den);
+                    Map.insert(newSym, v, acc)
+            }, Map#{}, facts))
+        case Join(d1, d2) =>
+            Join(renamePredSymsInternal(d1, mapping), renamePredSymsInternal(d2, mapping))
+        case _ => bug!("PredSym error")
+    }
+
+    ///
+    /// Applies `fixing` to the `d` to restore the `PredSym`s to the original values.
+    ///
+    @Internal
+    pub def renamePredSymToFlixStyle(d: Datalog, fixing: FixInformation): Datalog = match d {
```

Comment:
having FlixStyle in the name made me a bit confused - "what is Flix style?".

This function is just applying the fixing mapping to the predSyms of `d` right? So this function should just be called `renamePredSyms` and then its usage can be commented in the doc comment if needed